### PR TITLE
rpcPrincipalLookupV1 now supports findByTag()

### PR DESCRIPTION
**Pull request by ilaflamme-va**

### CODE REVIEW

File: charon-api-v1/src/main/java/gov/va/api/lighthouse/charon/api/v1/RpcPrincipalLookupV1.java [hunk 1]
```diff
@@ -19,6 +19,37 @@
...
+  /** Return the principal for a given RPC name at a given vista site. */
+  public Optional<RpcPrincipalV1> findByNameAndSite(String rpcName, String site) {
+    if (isBlank(rpcName) || isBlank(site)) {
+      return Optional.empty();
+    }
+    return Optional.ofNullable(findByName(rpcName).get(site));
+  }
+
+  /** Find all principals with the matching tag. */
+  public Map<String, RpcPrincipalV1> findByTag(String tag) {
```
Comment (jhulbert-va) [inline charon-api-v1/src/main/java/gov/va/api/lighthouse/charon/api/v1/RpcPrincipalLookupV1.java:34]: Is there a reason `findByTag` and `findEntriesByTag` are separate?

File: charon-api-v1/src/main/java/gov/va/api/lighthouse/charon/api/v1/RpcPrincipalLookupV1.java [hunk 2]
```diff
@@ -19,6 +19,37 @@
...
+    var entries = findEntriesByTag(tag);
+    return principalEntriesToMap(entries);
+  }
+
+  private List<RpcPrincipalsV1.PrincipalEntry> findEntriesByName(String rpcName) {
```
Comment (jhulbert-va) [inline charon-api-v1/src/main/java/gov/va/api/lighthouse/charon/api/v1/RpcPrincipalLookupV1.java:39]: Same with `findByName`.

File: charon-api-v1/src/main/java/gov/va/api/lighthouse/charon/api/v1/RpcPrincipalLookupV1.java (hunk 1 above)
Comment (ilaflamme-va) [reply charon-api-v1/src/main/java/gov/va/api/lighthouse/charon/api/v1/RpcPrincipalLookupV1.java:34]: To make the public method as clean and readable as possible. (it was like this already, I just continued the pattern)

Comment (ilaflamme-va) [reply charon-api-v1/src/main/java/gov/va/api/lighthouse/charon/api/v1/RpcPrincipalLookupV1.java:34]: I find that this
```
  /** Find all principals with the matching tag. */
  public Map<String, RpcPrincipalV1> findByTag(String tag) {
    var entries = findEntriesByTag(tag);
    return principalEntriesToMap(entries);
  }
 ```
 very clearly conveys intent to humans.

File: charon-api-v1/src/main/java/gov/va/api/lighthouse/charon/api/v1/RpcPrincipalLookupV1.java [hunk 3]
```diff
@@ -19,6 +19,37 @@
...
+        .collect(toList());
+  }
+
+  private List<RpcPrincipalsV1.PrincipalEntry> findEntriesByTag(String tag) {
+    return rpcPrincipalsV1.entries().stream()
+        .filter(principalEntry -> principalEntry.tags().contains(tag))
+        .collect(toList());
+  }
+
+  private Map<String, RpcPrincipalV1> principalEntriesToMap(
```
Comment (jhulbert-va) [inline charon-api-v1/src/main/java/gov/va/api/lighthouse/charon/api/v1/RpcPrincipalLookupV1.java:51]: Nit: `mapPrincipalEntriesBySite`

**Review by jhulbert-va — APPROVED** [review]

File: charon-api-v1/src/main/java/gov/va/api/lighthouse/charon/api/v1/RpcPrincipalLookupV1.java (hunk 3 above)
Comment (ilaflamme-va) [reply charon-api-v1/src/main/java/gov/va/api/lighthouse/charon/api/v1/RpcPrincipalLookupV1.java:51]: I like that

**Review by jhulbert-va — APPROVED** [review]

**Review by bschofield-va — APPROVED** [review]

**Pull request merged**